src/cbass/cbass.clj

```diff
@@ -25,23 +25,29 @@
 (defn ^Table get-table [^Connection c ^String t-name]
   (.getTable c (TableName/valueOf t-name)))
 
-(defn result-key [rk]
-  (-> (String. ^bytes (key rk)) keyword))
+(defn result-key [k result-key-type]
+  (case result-key-type
+    :keyword (-> (String. ^bytes k) keyword)
+    :string (String. ^bytes k)
+    :bytes k
+    (throw (IllegalArgumentException. "result-key-type must be one of the following: [:bytes :string :keyword]"))))
 
-(defn result-value [rv]
-  (@unpack (val rv)))
+(defn result-value [v] (@unpack v))
 
 (defn latest-ts [^Result result]
   (let [cells (.rawCells result)]
     (->> (map #(.getTimestamp ^Cell %) cells)
          (apply max))))
 
-(defn hdata->map [^Result data]
+(defn hdata->map [^Result data {:keys [result-key-type with-ts-per-cell?]
+                                :or {result-key-type :keyword with-ts-per-cell? false}}]
   (when-let [r (.getRow data)]
-    (let [ts (latest-ts data)
-          results (for [kv (-> (.getNoVersionMap data) vals first)]
-                    (if-some [v (result-value kv)]
-                      [(result-key kv) v]))]
+    (let [cells (.rawCells data)
```

Comment:
I think that it would be easier for cbass maintainer to accept the PR if you put the code that deals with timestamp per cell separate from the current code.